qiskit/pulse/instruction_schedule_map.py

```diff
@@ -321,12 +283,14 @@ def remove(
         instruction = _get_instruction_string(instruction)
         qubits = _to_tuple(qubits)
         self.assert_has(instruction, qubits)
-        self._map[instruction].pop(qubits)
-        self._qubit_instructions[qubits].remove(instruction)
+
+        del self._map[instruction][qubits]
         if not self._map[instruction]:
-            self._map.pop(instruction)
+            del self._map[instruction]
+
+        self._qubit_instructions[qubits].remove(instruction)
         if not self._qubit_instructions[qubits]:
-            self._qubit_instructions.pop(qubits)
+            del self._qubit_instructions[qubits]
```

Comment:
This seems like a bit of an unrelated change, it's better to keep the diff limited to the logical change of the PR. The logic looks the same to me so it's not anything worth blocking over.

Reply:
Yeah strictly speaking this is unrelated to this change but I realized this logic is slower according to this [stackoverflow](https://stackoverflow.com/questions/24547223/is-del-or-pop-preferred-when-removing-elements-from-dicts). So I made this change to improve overall performance of inst map.